program/object-detection-armnn-tflite/includes/settings.h

```diff
@@ -0,0 +1,365 @@
+/*
+ * Copyright (c) 2018 cTuning foundation.
+ * See CK COPYRIGHT.txt for copyright details.
+ *
+ * SPDX-License-Identifier: BSD-3-Clause.
+ * See CK LICENSE.txt for licensing details.
+ */
+
+#ifndef DETECT_SETTINGS_H
+#define DETECT_SETTINGS_H
+
+#include <iostream>
+#include <thread>
+#include <stdlib.h>
+#include <vector>
+#include <fstream>
+#include <sstream>
+#include <map>
+#include <dirent.h>
+
+
+template<char delimiter>
+class WordDelimitedBy : public std::string {
+};
+
+template<char delimiter>
+std::istream &operator>>(std::istream &is, WordDelimitedBy<delimiter> &output) {
+    std::getline(is, output, delimiter);
+    return is;
+}
+struct FileInfo {
+    std::string name;
+    int width;
+    int height;
+};
+
+bool get_yes_no(std::string) ;
+bool get_yes_no(char *);
+std::string str_to_lower(std::string);
+std::string str_to_lower(char *);
+std::vector<std::string> *readClassesFile(std::string);
+float *readAnchorsFile(std::string, int&);
+
+inline std::string alter_str(std::string a, std::string b) { return a != "" ? a: b; };
+inline std::string alter_str(char *a, std::string b) { return a != nullptr ? a: b; };
+
+
+class Settings {
+public:
+    Settings() {
+
+        // Load settings
+        std::ifstream settings_file("env.ini");
+        if (!settings_file)
+            throw "Unable to open 'env.ini' file";
+        _verbose = get_yes_no(getenv("VERBOSE"));
+        if (_verbose) std::cout << "Settings from 'env.ini' file:" << std::endl;
+        std::map<std::string, std::string> settings_from_file;
+        for (std::string s; !getline(settings_file, s).fail();) {
+            if (_verbose) std::cout << '\t' << s << std::endl;
+            std::istringstream iss(s);
+            std::vector<std::string> row((std::istream_iterator<WordDelimitedBy<'='>>(iss)),
+                                         std::istream_iterator<WordDelimitedBy<'='>>());
+            if (row.size() == 1)
+                settings_from_file.emplace(row[0], "");
+            else
+                settings_from_file.emplace(row[0], row[1]);
+        }
+
+        std::string model_dataset_type = settings_from_file["MODEL_DATASET_TYPE"];
+        if (model_dataset_type != "coco") {
+            throw ("Unsupported model dataset type: " + model_dataset_type);
+        }
+
+        std::string nms_type = alter_str(getenv("USE_NMS"), "no");
+        if (str_to_lower(nms_type) == "no") {
+            _graph_file = std::string(getenv("CK_ENV_TENSORFLOW_MODEL_TFLITE_GRAPH_NO_NMS"));
+        } else {
+            std::cout << std::endl << "ERROR: Unsupported USE_NMS type - " << nms_type << std::endl;
+            exit(-1);
+        }
+        _graph_file = std::string(getenv("CK_ENV_TENSORFLOW_MODEL_ROOT")) + "/" + _graph_file;
```

Comment:
`/` is *nix-specific delimiter. Make more portable?

Reply:
Fixed